src/Plugin/QuickForm/SaatenUnionSpraying.php

```diff
@@ -0,0 +1,427 @@
+    $form['weather'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Weather'),
+      '#description' => $this->t('The dominant weather conditions during spraying.'),
+      '#options' => $weather_types_options,
+      '#multiple' => TRUE,
+      '#required' => TRUE,
+    ];
+
+    // $form['pressure'] = [
+    //   '#type' => 'number',
+    //   'title' => $this->t('Pressure'),
+    //   'description' => $this->t('The water pressure used when applying the product, where relevant.'),
+    //   'measure' => ['#value' => 'pressure'],
+    //   'units' => ['#value' => 'bar'],
+    // ];
+
+    $equipments = $this->getActiveEquipmentLabels();
+
+    $form['equipment'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Equipment'),
+      '#description' => $this->t('Select the tractor, sprayer and nozzle equipment assets used for this spraying'),  
+      '#options' => $equipments,
+      '#required' => TRUE,
+      '#multiple' => TRUE,
+    ];
+
+    $form['justification_target'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Justification/Target'),
+      '#description' => $this->t('The reason the operation is necessary, and any target pest(s) where applicable.'),
+      '#required' => TRUE,
+    ];
+
+    $form['plant_growth_stage'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Plant growth stage'),
+      '#description' => $this->t('The plant growth stage when the product was applied.'),
+      '#required' => FALSE,
+    ];
+
+    $form['equipment_rinsed'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment triple rinsed'),
+      '#description' => $this->t('Select if the equipment was triple rinsed after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    $form['equipment_clear'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment All Clear'),
+      '#description' => $this->t('Select if the equipment all clear after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    $form['equipment_washed'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Equipment Clear Washed'),
+      '#description' => $this->t('Select if the equipment was clear washed after the job was completed.'),
+      '#return_value' => 'Yes',
+    ];
+
+    return $form;
+  }
+
+  /**
+  * Submit handler for Saaten Union Spraying Quick Form.
+  */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+
+    // Create a new log entity.
+    $log = $this->createLog([
+      'type' => "input",
+      'name' => $form_state->getValue('log_name'),
+      'timestamp' =>  $form_state->getValue('date_start'),
+      'status' => $form_state->getValue('status'),
+      'flag' => $form_state->getValue('flag'),
+    ]);
+
+    // Save the log entity.
+    $log->save();
+
+    // Display a success message.
+    $this->messenger()->addMessage($this->t('The spraying log has been saved.'));
```

Comment:
This is unnecessary because it is provided by the `QuickLogTrait::createLog` method. See https://github.com/farmOS/farmOS/blob/d8ff8e1cbdf201b97a322cc60c9fd44133743e26/modules/core/quick/src/Traits/QuickLogTrait.php#L73-L81

Reply:
I have made these updates